### PR TITLE
Filter Partitions Processed by Default

### PR DESCRIPTION
**Pull request by dfjxs**
* Filtering partitions processed by default based on filesystem and size
* Adding task config options to control which partitions are processed

Fixes: #729

### CODE REVIEW

**Comment by dfjxs** [comment]
@aarontp Tests have passed, but since the local e2e tests use `test_data/artifact_disk.dd` the partition will be filtered out by default (based on size).

```
2022-08-19 04:55:59 [INFO] Scanning [/evidence/artifact_disk.dd] for partitions
2022-08-19 04:55:59 [INFO] Marking partition /p1 unimportant (size 19906048 bytes)
2022-08-19 04:55:59 [INFO] Not processing unimportant partition DiskPartition:/p1 due to task config
2022-08-19 04:55:59 [INFO] Scanning of [/evidence/artifact_disk.dd] is complete
2022-08-19 04:55:59 [INFO] Found 1 partition(s) in [/evidence/artifact_disk.dd] Processing 0 partitions:
```

We should probably run the e2e tests with the task config `process_unimportant: True`

File: turbinia/workers/partitions.py [hunk 1]
```diff
@@ -130,7 +141,27 @@ def _ProcessPartition(self, path_spec):
...
+    # Is partition important based on filesystem?
+    if path_spec.type_indicator not in ('APFS', 'EXT', 'HFS', 'NTFS', 'TSK',
+                                        'XFS'):
+      important = False
+      log.info(
+          'Marking partition {0:s} unimportant (filesystem {1:s})'.format(
+              location, path_spec.type_indicator))
+
+    # Is partition important based on size? (100M or larger)
+    if partition_size and partition_size < 104857600:
```
Comment (aarontp) [inline turbinia/workers/partitions.py:153]: Maybe make this threshold one of the task config variables?

File: turbinia/workers/partitions.py [hunk 2]
```diff
@@ -130,7 +141,27 @@ def _ProcessPartition(self, path_spec):
 
       child_path_spec = child_path_spec.parent
 
+    # Is partition important based on filesystem?
+    if path_spec.type_indicator not in ('APFS', 'EXT', 'HFS', 'NTFS', 'TSK',
+                                        'XFS'):
+      important = False
+      log.info(
+          'Marking partition {0:s} unimportant (filesystem {1:s})'.format(
```
Comment (aarontp) [inline turbinia/workers/partitions.py:149]: Might be good to mention why it was marked unimportant here (e.g. was a non-standard FS type {$foo}).

File: turbinia/workers/partitions.py [hunk 3]
```diff
@@ -130,7 +141,27 @@ def _ProcessPartition(self, path_spec):
...
+              location, path_spec.type_indicator))
+
+    # Is partition important based on size? (100M or larger)
+    if partition_size and partition_size < 104857600:
+      important = False
+      log.info(
+          'Marking partition {0:s} unimportant (size {1!s} bytes)'.format(
```
Comment (aarontp) [inline turbinia/workers/partitions.py:156]: Could also be a little more explicit here and say "size foo < size bar".

**Review by aarontp — APPROVED** [review]
LGTM, just some small suggestions.

I also filed https://github.com/google/turbinia/issues/1118 for the e2e tests to process "unimportant" partitions.

File: turbinia/workers/partitions.py (hunk 1 above)
Comment (dfjxs) [reply turbinia/workers/partitions.py:153]: Done

File: turbinia/workers/partitions.py (hunk 3 above)
Comment (dfjxs) [reply turbinia/workers/partitions.py:156]: Done

**Comment by dfjxs** [comment]
Also added a recipe to local e2e tests which fixes: #1118

**Pull request merged**